client/src/pages/HomePage/view.jsx

```diff
@@ -4,6 +4,9 @@ import PageLayout from '@components/PageLayout';
 import PublishTool from '@containers/PublishTool';
 
 class HomePage extends React.Component {
+  componentWillUnmount () {
+    this.props.clearFile();
```

Comment:
Why does this have to happen here? Does this have to happen for any component that includes `<PublishTool />`? If so, this should be avoided.

Reply:
Nice catch/good point! Current speech only clears the file for file errors, reset, publish success, and cancel publish, so two features using the same `state.publish` were stepping on each others toes

My new approach uses `isUpdate`

When the publish page mounts: if `isUpdate` then `clearFile`

When edit page mounts: if `!isUpdate` then `clearFile, setUpdateTrue, setMetadata`

This way a user can start a publish or update, leave the page, and come back to where they were, like current speech, unless they go to the other (Edit/Publish) page, in which case it starts fresh with publish or update

[code](https://github.com/lbryio/spee.ch/pull/595/commits/7c4323ba648fea1e8e2b361a17654b10ae66ff93)

Reply:
:+1: